copy/copy.go

```diff
@@ -92,14 +95,22 @@ type Options struct {
 	ReportWriter     io.Writer
 	SourceCtx        *types.SystemContext
 	DestinationCtx   *types.SystemContext
+	ProgressInterval time.Duration                 // time to wait between reports to signal the progress channel
+	Progress         chan types.ProgressProperties // Reported to when ProgressInterval has arrived for a single artifact+offset.
```

Comment:
AFAICS this should actually be a send-only channel. The code calling `copy.Image` needs a bidirectional channel, but `copy.Image` will only be sending to it, and the rest of the code calling `copy.Image` will only be receiving from it.